hooks/post/31-mongo_syspath.rb

```diff
@@ -0,0 +1,23 @@
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def install_syspaths
+  `rpm -qa | grep rh-mongodb34-syspaths`
+  if $?.success?
+    logger.debug 'rh-mongodb34-syspaths already installed, skipping.'
+  else
+    logger.info 'rh-mongodb34-syspaths not present, installing.'
+    Kafo::Helpers.execute('yum install -y -q rh-mongodb34-syspaths')
+  end
+end
+
+# Fail if we are not using the Katello/Foreman Proxy Content scenario.
+katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
+foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
+if katello || foreman_proxy_content
+  install_syspaths
+else
+  fail_and_exit 'MongoDB storage engine upgrade is not currently supported for the chosen scenario.'
```

Comment:
Should we fail here or just warn that a scenario that is not supported was detected and thus mongodb-syspaths was not installed? The likely hood of hitting this hook under that condition is slim but I think thats cleaner than hard fail.

Reply:
+1 to warn

Reply:
IMHO it should be quiet. The certificate generation script also calls hooks so I think that failing here breaks that functionality.